RFC-0039-cuda-support.md

```diff
@@ -0,0 +1,197 @@
+
+2. Evaluate if we have all packages for update
+  When: As soon as Update determined to be necessary. Start by creating RFC [issue](https://github.com/pytorch/pytorch/issues/145544) with possible CUDA matrix to support for next release.
+  Goal: Make sure everything is available to perform complete upgrade of CUDA and dependencies
+
+3. Update CUDA in CD (this is necessary condition to qualify for CUDA version be released as Experimental)
+  When: Evaluate if we have all packages for update is complete
+  Goal: Make sure all binaries Linux/Windes on wheels and conda platforms are produced on nightly
+
+4. Update CUDA in CI
+  When: Update CUDA in CD is complete
+  Goal: Make sure CI jobs tests on new CUDA versions, all failing tests should be identified and reported
+
+5. Run benchmarks to compare new experimental CUDA version to current stable CUDA version
+  When: Update CUDA in CD  is complete
+  Goal: Make sure we identify all performance regressions
+
+
+### Detailed Process of Deprecating CUDA version
+
+1. Evaluate deprecation of legacy CUDA version from CI/CD
+When: We completed CUDA update and previous experimental CUDA version is qualified to be stable and we have 3 supported versions (legacy, stable and experimental). Start by creating RFC [issue](https://github.com/pytorch/pytorch/issues/147383) to deprecate legacy CUDA version from CI/CD.
+Goal: Make sure we support 2 versions of CUDA, supporting 3 versions can be an exception for certain release where we need to keep legacy version
+
+2. Deprecate legacy CUDA version from CI/CD
+When: Evaluate deprecation of legacy CUDA version from CI/CD is complete
+Goal: Support for legacy CUDA versions is dropped, starting from PyTorch Domain Libraries and then in PyTorch core. First we drop CD support and then CI support.
+
+
+# CUDA/CUDNN Upgrade Runbook
+
+So you wanna upgrade PyTorch to support a new CUDA? Follow these steps in order! They are adapted from previous CUDA upgrade processes.
+
+## 0. Before starting upgrade process
+
+### A. Currently Supported CUDA and CUDNN Libraries
+Here is the supported matrix for CUDA and CUDNN (versions can be looked up in [here](https://github.com/pytorch/pytorch/blob/main/.ci/docker/common/install_cudnn.sh) . Architecture support can be found [here](https://github.com/pytorch/pytorch/blob/main/.ci/manywheel/build_cuda.sh))
+
+| CUDA | CUDNN | architectures supported | additional details |
+| --- | --- | --- | --- |
+| 11.8.0 | 9.1.0.70 | Kepler(3.7), Maxwell(5.0), Pascal(6.0), Volta(7.0), Turing(7.5), Ampere(8.0, 8.6), Hopper(9.0) | Legacy CUDA Release |
+| 12.6.3 | 9.5.1.17 | Maxwell(5.0), Pascal(6.0), Volta(7.0), Turing(7.5), Ampere(8.0, 8.6), Hopper(9.0)  | Stable CUDA Release |
+| 12.8.0 | 9.7.1.26 | Turing(7.5), Ampere(8.0, 8.6), Hopper(9.0), Blackwell(10.0, 12.0+PTX)  | Latest CUDA Release |
+|        | 9.8.0.87 | Turing(7.5), Ampere(8.0, 8.6), Hopper(9.0), Blackwell(10.0, 12.0+PTX)  | Latest CUDA Nightly |
```

Comment:
We should add this matrix to the install matrix on https://pytorch.org/get-started/locally/

Reply:
Lets add it to RELEASE.md as well